src/test/java/com/theelixrlabs/healthcare/service/DoctorServiceTest.java

```diff
@@ -0,0 +1,105 @@
+
+/**
+ * Unit tests for the DoctorService class.
+ * This class contains test cases to verify the behavior of the methods in the DoctorService class.
+ */
+public class DoctorServiceTest {
+    @Mock
+    private DoctorRepository doctorRepository;
+    @Mock
+    private Validator validator;
+    @Mock
+    private MessageUtil messageUtil;
+    @InjectMocks
+    private DoctorService doctorService;
+    private List<DoctorModel> doctorModelList;
+
+    /**
+     * Initializes the test data and mocks before each test is run.
+     */
+    @BeforeEach
+    public void setUp() {
+        MockitoAnnotations.openMocks(this);
+        doctorModelList = new ArrayList<>();
+        // Adding first doctor model
+        DoctorModel doctorModel1 = DoctorModel.builder()
+                .id(UUID.randomUUID())
+                .firstName("John")
+                .lastName("Doe")
+                .department("Cardiology")
+                .aadhaarNumber("2365 5896 4521")
+                .build();
+        doctorModelList.add(doctorModel1);
+        // Adding second doctor model
+        DoctorModel doctorModel2 = DoctorModel.builder()
+                .id(UUID.randomUUID())
+                .firstName("Johnson")
+                .lastName("Chris")
+                .department("Java")
+                .aadhaarNumber("5678 8585 6565")
+                .build();
+        doctorModelList.add(doctorModel2);
+    }
+
+    /**
+     * This test verifies that the service correctly retrieves and maps doctor models to DTOs when there are matches in the repository.
+     *
+     * @throws Exception if any error occurs during the test
+     */
+    @Test
+    public void getDoctorsByName_Success() throws Exception {
+        String doctorName = doctorModelList.get(0).getFirstName();
+        when(doctorRepository.searchByDoctorName(doctorName)).thenReturn(doctorModelList);
+        List<DoctorDto> result = doctorService.getDoctorsByName(doctorName);
+        assertNotNull(result);
+        assertEquals(2, result.size());
+        // Checking the first doctor
+        assertEquals("John", result.get(0).getFirstName());
+        assertEquals("Doe", result.get(0).getLastName());
+        assertEquals("Cardiology", result.get(0).getDepartment());
+        assertEquals("2365 5896 4521", result.get(0).getAadhaarNumber());
+        // Checking the second doctor
+        assertEquals("Johnson", result.get(1).getFirstName());
+        assertEquals("Chris", result.get(1).getLastName());
+        assertEquals("Java", result.get(1).getDepartment());
+        assertEquals("5678 8585 6565", result.get(1).getAadhaarNumber());
+    }
+
+    /**
+     * This test verifies that the service throws a DoctorNotFoundException with the correct message when no matching doctors are found in the repository.
+     *
+     * @throws Exception if any error occurs during the test
+     */
+    @Test
+    public void getDoctorsByName_Failure() throws Exception {
```

Comment:
```suggestion
    public void getDoctorsByName_ReturnsNotFoundException() throws Exception {
```

Reply:
changed the method name to getDoctorsByName_ReturnsNotFoundException()